vizanti_server/public/templates/satelite/satelite_script.js

```diff
@@ -187,7 +205,7 @@ async function drawTiles(){
 
 		// Convert the corners to tile coordinates
 		const cornerTileCoords = cornerCoords.map((coord) =>
-			navsat.coordToTile(coord.longitude, coord.latitude, zoomLevel)
+			navsat.coordToTile(coord.longitude, coord.latitude, tempZoomLevel)
 		);
 
 		// Calculate the range of tiles to cover the screen
```

Comment:
This part has always been a bit broken, as it doesn't quite calculate the correct range of tiles if you move further away from the fixed frame origin. When moving horizontally it starts to chop off the top and bottom rows, and the reverse when moving vertically.

While zooming out doesn't seem to make it much worse, given that we're doing this rework it would be a great to have a fresh pair of eyes to look at it, I could never quite figure out what was going wrong here math-wise (I think the error might be in [this transform](https://github.com/MoffKalast/vizanti/blob/e11d71a8fe250af00d7372abd49d80e49b5a1787/vizanti_server/public/templates/satelite/satelite_script.js#L192)).

Reply:
> This part (tile screen coverage) has always been a bit broken, as it doesn't quite calculate the correct range of tiles if you move further away from the fixed frame origin. When moving horizontally it starts to chop off the top and bottom rows, and the reverse when moving vertically.
> 
> While zooming out doesn't seem to make it much worse, given that we're doing this rework it would be a great to have a fresh pair of eyes to look at it, I could never quite figure out what was going wrong here math-wise (I think the error might be in [this transform](https://github.com/MoffKalast/vizanti/blob/e11d71a8fe250af00d7372abd49d80e49b5a1787/vizanti_server/public/templates/satelite/satelite_script.js#L192)).

I will look into it more closely then. I am trying to understand how it works, and I am curious why you created but never use the `tileSizeInDegrees` function from **vizanti_server/public/js/modules/navsat/js**,because it seems that only `tileSizeInMeters` is used for calculations.

Reply:
Hmm, I think `tileSizeInDegrees` and `tileSizeInMeters` are both from a specification I found somewhere at the time, the navsat.js file is intended to be just a general library module for navsat related things, so I kept both in case the second one ever comes in handy.

Speaking of navsat.js, seems like adding a queue reset upon layer change (so it starts loading the new layer immediately) makes it a lot more responsive, e.g:

navsat.js
```js
async clear_queue(keyurl){
    this.queue = new Set();
    this.queue_history = new Set();
}
```
satelite_script.js
```js
const frame = tf.absoluteTransforms[map_fix.header.frame_id]; 
let tempZoomLevel = Math.round(Math.log2(view.scale)+17);
tempZoomLevel = clamp(tempZoomLevel, 7, 19); 

if(tempZoomLevel != zoomLevel){
  navsat.clear_queue();
  zoomLevel = tempZoomLevel;
}
```

Also changing [the throttle param from 500 to 200](https://github.com/MoffKalast/vizanti/blob/d68fbbc21c2d43b103c19c710fe50d8ee0b6844f/public/js/modules/navsat.js#L59) seems to help a lot in loading speed.

I've looked into the empty flashing issue a bit, but so far I can't explain it, even removing the `clearRect` doesn't seem to help. It's like rendering stalls entirely or something.

Reply:
Empty flashing you mean behavior like in this video? [vizanti_test_1](https://youtu.be/ZycWOYqUUUs) in 6s and lather when I move the map and it disappears in front of the user?

Reply:
That's the culling of tiles based on that wrong calculation of what's on screen and what isn't. Seems like rotation of the fix tf_frame affects it a lot as well, there's definitely something up with the transform part.

What I'm seeing actually doesn't happen in the video at all, it's a split second moment where no tiles render and only the background colour shows. Happens when the zoom level changes. It might be browser specific I guess.

Reply:
There's an older discussion regarding the tf link rotation for more context on that, in case it helps: https://github.com/MoffKalast/vizanti/issues/52